library/src/main/java/eu/kanade/tachiyomi/source/online/ResolvableSource.kt

```diff
@@ -1,27 +1,43 @@
 package eu.kanade.tachiyomi.source.online
 
 import eu.kanade.tachiyomi.source.Source
+import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
 
 /**
- * A source that may handle opening an SManga for a given URI.
+ * A source that may handle opening an SManga or SChapter for a given URI.
  *
  * @since extensions-lib 1.5
  */
-@Suppress("unused")
 interface ResolvableSource : Source {
 
     /**
-     * Whether this source may potentially handle the given URI.
+     * Returns what the given URI may open.
+     * Returns [UriType.Unknown] if the source is not able to resolve the URI.
      *
      * @since extensions-lib 1.5
      */
-    fun canResolveUri(uri: String): Boolean
+    fun getUriType(uri: String): UriType
```

Comment:
Could probably do `fun resolveUrl(url: String): Pair<SManga, SChapter?>`

Reply:
that's just how it is currently in the app, but sure can be changed